plugins/scaffolder-backend/src/service/router.ts

```diff
@@ -227,7 +211,17 @@ export async function createRouter(
       });
 
   actionsToRegister.forEach(action => actionRegistry.register(action));
-  workers.forEach(worker => worker.start());
+
+  const worker = await TaskWorker.create({
```

Comment:
@punkle this doesn't impact your setup does it? I'm thinking you'll also gladly have the same switch from multiple workers to a single worker with capacity for multiple jobs.

Reply:
Will this create a worker to run with the API? Is it possible to have a switch to turn this off? We dont want scaffolder templates to run along side the API process?

Reply:
Ah right, that's the customization you needed. I guess the simplest route is the one we're already at, i.e. if `concurrentTasksLimit` is set to `0`, then we won't bother creating a worker to execute tasks.

Reply:
I reverted the changes I made regarding multiple TaskWorker creation. Now the `router` file stays largely unchanged (except the default TaskWorker counter is back to 1). Maybe the work of coalescing the `concurrentTasksLimit` and `taskWorkers` option can be done in another PR.